File: java/number1.java
public class Number1 {
    public static Object startPosFinder(int m, int n, String s) {
        int x = 0, y = 0;
        int rX = 0, lX = 0, uY = 0, dY = 0;

        for (char i : s.toCharArray()) {
            switch (i) {
                case 'D':
                    y += 1;
                    if (y > dY) dY = y;
                    break;
                case 'U':
                    y -= 1;
                    if (y < uY) uY = y;
                    break;
                case 'L':
                    x -= 1;
                    if (x < lX) lX = x;
                    break;
                case 'R':
                    x += 1;
                    if (x > rX) rX = x;
                    break;
            }
        }

        if (rX - lX >= m || dY - uY >= n) {
            return "No";
        } else {
            return new int[]{1 - lX, 1 - uY};
        }
    }

    public static void main(String[] args) {
        int m = 3, n = 3;
        String s = "ULDDRRUU";

        Object result = startPosFinder(m, n, s);
        if (result instanceof String) {
            System.out.println(result);
        } else {
            int[] position = (int[]) result;
            System.out.println("[" + position[0] + ", " + position[1] + "]");
        }
    }
}

File: kt/number1.kt
val m = 3
val n = 3
val s = "ULDDRRUU"

fun startPosFinder(m: Int, n: Int, s: String): Any {
    var x = 0
    var y = 0
    var rX = 0
    var lX = 0
    var uY = 0
    var dY = 0

    for (i in s) {
        when (i) {
            'D' -> {
                y += 1
                if (y > dY) dY = y
            }
            'U' -> {
                y -= 1
                if (y < uY) uY = y
            }
            'L' -> {
                x -= 1
                if (x < lX) lX = x
            }
            'R' -> {
                x += 1
                if (x > rX) rX = x
            }
        }
    }

    return if (rX - lX >= m || dY - uY >= n) {
        "No"
    } else {
        Pair(1 - lX, 1 - uY)
    }
}

fun main() {
    println(startPosFinder(m, n, s))
}